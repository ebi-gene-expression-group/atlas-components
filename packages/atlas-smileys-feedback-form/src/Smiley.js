import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Smiley = styled.div`
  transition: all 0.5s;
  text-align: center;
  cursor: pointer;
  ::before {
    font-size: 2.5rem;
    transition: inherit;
    content: '${props => props.emoji}';
    opacity: ${props => props.selected ? 1.0 : 0.2};
  }
  :hover::before {
    opacity: 1.0
  }

  ::after {
    font-size: 1rem;
    transition: inherit;
    content: '${props => props.label}';
    opacity: ${props => props.selected ? 1.0 : 0.0};
  }
  :hover::after {
    opacity: 1.0
  }

`

Smiley.propTypes = {
  emoji: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  selected: PropTypes.bool.isRequired
}

export default Smiley
