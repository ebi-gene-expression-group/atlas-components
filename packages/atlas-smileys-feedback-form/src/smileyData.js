const smileyData = [
  {
    emoji: `😞`,
    label: `Terrible`,
    score: 1
  },
  {
    emoji: `😔`,
    label: `Bad`,
    score: 2
  },
  {
    emoji: `😐`,
    label: `Okay`,
    score: 3
  },
  {
    emoji: `😁`,
    label: `Good`,
    score: 4
  },
  {
    emoji: `😆`,
    label: `Great`,
    score: 5
  }
]

export default smileyData
