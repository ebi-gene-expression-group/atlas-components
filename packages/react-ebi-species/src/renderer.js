"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
