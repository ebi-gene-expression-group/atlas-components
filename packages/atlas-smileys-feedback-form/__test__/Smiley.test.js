import React from 'react'
import renderer from 'react-test-renderer'

import smileyData from '../src/smileyData'
import Smiley from '../src/Smiley'

describe(`Smiley`, () => {
  // Unfortunately, Enzyme doesn’t support pseudoelement selection so we can’t test any property of the emoji or the
  // label, snapshots will need to suffice for now
  test(`matches snapshot`, () => {
    const tree = renderer.create(
      <div>
        { smileyData.map((smiley, idx) => <Smiley key={idx} {...smiley} selected={true} onClick={() => {}}/>) }
        { smileyData.map((smiley, idx) => <Smiley key={idx} {...smiley} selected={false} onClick={() => {}}/>) }
      </div>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
