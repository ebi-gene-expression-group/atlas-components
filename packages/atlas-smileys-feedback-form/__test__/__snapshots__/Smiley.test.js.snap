// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smiley matches snapshot 1`] = `
<div>
  <div
    className="sc-bdVaJa eUBkZK"
    label="Terrible"
    onClick={[Function]}
    selected={true}
  />
  <div
    className="sc-bdVaJa ljUdrQ"
    label="Bad"
    onClick={[Function]}
    selected={true}
  />
  <div
    className="sc-bdVaJa bGpVwd"
    label="Okay"
    onClick={[Function]}
    selected={true}
  />
  <div
    className="sc-bdVaJa dZFsbh"
    label="Good"
    onClick={[Function]}
    selected={true}
  />
  <div
    className="sc-bdVaJa iSjcWB"
    label="Great"
    onClick={[Function]}
    selected={true}
  />
  <div
    className="sc-bdVaJa jyeDiH"
    label="Terrible"
    onClick={[Function]}
    selected={false}
  />
  <div
    className="sc-bdVaJa dpPsPz"
    label="Bad"
    onClick={[Function]}
    selected={false}
  />
  <div
    className="sc-bdVaJa iTGzAl"
    label="Okay"
    onClick={[Function]}
    selected={false}
  />
  <div
    className="sc-bdVaJa gNkLYR"
    label="Good"
    onClick={[Function]}
    selected={false}
  />
  <div
    className="sc-bdVaJa fksfqK"
    label="Great"
    onClick={[Function]}
    selected={false}
  />
</div>
`;
